### PR TITLE
app/vmalert: adds idleConnTimeout flags and retry trivial network errors

### PR DESCRIPTION
**Pull request by f41gh7**
* "*.idleConnTimeout" flags must reduce probability of `write: broken pipe` and `read: connection reset by peer` errors Those errors may occur if remote server closes TCP socket for connection, while it's still exist at client.
* single time retries for `write: broken pipe` and `read: connection reset by peer` must handle a case for incorrectly configured timeouts at middleware proxies, mitigate minor network issues.

https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5661

### Describe Your Changes

Please provide a brief description of the changes you made. Be as specific as possible to help others understand the purpose and impact of your modifications.

### CODE REVIEW

File: docs/vmauth.md [hunk 1]
```diff
@@ -1206,6 +1206,8 @@ See the docs at https://docs.victoriametrics.com/vmauth/ .
      Whether to use proxy protocol for connections accepted at the corresponding -httpListenAddr . See https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt . With enabled proxy protocol http server cannot serve regular /metrics endpoint. Use -pushmetrics.url for metrics pushing
      Supports array of values separated by comma or specified via multiple flags.
      Empty values are set to false.
+  -idleConnTimeout duration
```
Comment (hagen1778) [inline docs/vmauth.md:1209]: I don't see this flag supported by vmauth.

File: app/vmalert/datasource/init.go [hunk 2]
```diff
@@ -57,8 +57,9 @@ var (
...
 		"See more details at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/1257")
-	maxIdleConnections = flag.Int("datasource.maxIdleConnections", 100, `Defines the number of idle (keep-alive connections) to each configured datasource. Consider setting this value equal to the value: groups_total * group.concurrency. Too low a value may result in a high number of sockets in TIME_WAIT state.`)
-	disableKeepAlive   = flag.Bool("datasource.disableKeepAlive", false, `Whether to disable long-lived connections to the datasource. `+
+	maxIdleConnections    = flag.Int("datasource.maxIdleConnections", 100, `Defines the number of idle (keep-alive connections) to each configured datasource. Consider setting this value equal to the value: groups_total * group.concurrency. Too low a value may result in a high number of sockets in TIME_WAIT state.`)
+	idleConnectionTimeout = flag.Duration("datasource.idleConnTimeout", 50*time.Second, `Defines a duration for idle (keep-alive connections) to exist. Consider settings this value less to the value of "-http.idleConnTimeout". It must prevent possible "write: broken pipe" and "read: connection reset by peer" errors.`)
```
Comment (hagen1778) [inline app/vmalert/datasource/init.go:61]:
```suggestion
	idleConnectionTimeout = flag.Duration("datasource.idleConnTimeout", 50*time.Second, `Defines a duration for idle (keep-alive connections) to exist. Consider setting this value less than "-http.idleConnTimeout". It must prevent possible "write: broken pipe" and "read: connection reset by peer" errors.`)
```

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -50,6 +50,8 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
...
 * FEATURE: [vmagent](https://docs.victoriametrics.com/vmagent.html): add labels `path` and `url` to metrics `vmagent_remotewrite_push_failures_total` and `vmagent_remotewrite_samples_dropped_total`. Now number of failed pushes and dropped samples can be tracked per `-remoteWrite.url`.
 * FEATURE: [stream aggregation](https://docs.victoriametrics.com/stream-aggregation/): add [rate_sum](https://docs.victoriametrics.com/stream-aggregation/#rate_sum) and [rate_avg](https://docs.victoriametrics.com/stream-aggregation/#rate_avg) aggregation outputs.
+* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert/): adds `datasource.idleConnTimeout`, `remoteWrite.idleConnTimeout` and `remoteRead.idleConnTimeout` flags. Which must reduce probability of `broken pipe` and `connection reset by peer` errors. See this [issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5661) for details.
```
Comment (hagen1778) [inline docs/CHANGELOG.md:53]:
```suggestion
* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert/): add `datasource.idleConnTimeout`, `remoteWrite.idleConnTimeout` and `remoteRead.idleConnTimeout` flags. These flags are set to 50s by default and should reduce the probability of `broken pipe` or `connection reset by peer` errors in vmalert logs. See this [issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5661) for details.
```

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -50,6 +50,8 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
...
+* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert/): Adds request retry for trivial network errors, such as `broken pipe` and `connection reset`. See this [issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5661) for details.
```
Comment (hagen1778) [inline docs/CHANGELOG.md:54]:
```suggestion
* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert/): add auto request retry for trivial network errors, such as `broken pipe` and `connection reset` for requests to `remoteRead`, `remoteWrite` and `datasource` URLs. See this [issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5661) for details.
```

**Review by hagen1778 — APPROVED** [review]
LGTM
a few comments, though

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.102.0-rc1 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.102.0-rc1).

**Pull request merged**